runner/ollamarunner/runner.go

```diff
@@ -564,25 +556,43 @@ type CompletionResponse struct {
 	Timings Timings `json:"timings"`
 }
 
-func getSamplers(_ CompletionRequest) []sample.Sampler {
-	// TODO(jessegross): Waiting for sampling code
-
-	/*samplingParams.TopK = req.TopK
-	samplingParams.TopP = req.TopP
-	samplingParams.MinP = req.MinP
-	samplingParams.TypicalP = req.TypicalP
-	samplingParams.Temp = req.Temperature
-	samplingParams.RepeatLastN = req.RepeatLastN
-	samplingParams.PenaltyRepeat = req.RepeatPenalty
-	samplingParams.PenaltyFreq = req.FrequencyPenalty
-	samplingParams.PenaltyPresent = req.PresencePenalty
-	samplingParams.Mirostat = req.Mirostat
-	samplingParams.MirostatTau = req.MirostatTau
-	samplingParams.MirostatEta = req.MirostatEta
-	samplingParams.Seed = uint32(req.Seed)
-	samplingParams.Grammar = req.Grammar*/
-
-	return []sample.Sampler{sample.Greedy()}
+func newSampler(req CompletionRequest) sample.Sampler {
```

Comment:
This is fine, but I would verify that we even need this "constructor" here and can just do with creating a `sample.Sampler` with parameters directly:

```
sampler := sample.Sampler{
  Temperature: 0.6
  ... other sampling parameters ...
}
```

Reply:
Note: if there's a default "chaining" order, should the sample package maybe encode this?

Reply:
I had initially set it up like that but with discussions it seemed like we wanted to keep the sample package really simple and let the consumer of the package decide how they wanted to use it. I am happy to go back to that and structure it more in the package. cc: @mxyng